code/game/objects/items/weapons/RCD.dm

```diff
@@ -286,11 +288,15 @@
 				message_admins("RCD Door HREF exploit attempted by [key_name(usr)]!")
 				return FALSE
 			door_type = new_door_type
+			var/obj/machinery/door/airlock/proto = new door_type(null)
+			if(proto.glass)
+				airlock_glass = TRUE
+			else
+				airlock_glass = FALSE
+			qdel(proto)
```

Comment:
You can save having to create a new airlock by using `initial()`. Can do something like:
```dm
var/obj/machinery/door/airlock/proto = door_type
airlock_glass = initial(proto.glass)
```